tests/html_glimmer/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -0,0 +1,493 @@
+{{@greeting}}
+,
+{{@name}}
+!
+<button onclick={{action next}}>
+  Next
+</button>
+`;
+
+exports[`concat-statement.hbs 1`] = `
+<div class="hello {{if goodbye true}}">
+  Hello
+</div>
+
+<div class="hello {{if goodbye true}} {{if goodbye false}} {{if goodbye true}} {{if goodbye false}} {{if goodbye true}}">
+  Hello
+</div>
+
+<a href="/{{url}}/{{url}}"></a>
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+<div class="hello{{if goodbye true}}">
+  Hello
+</div>
+<div
+  class="hello
+    {{if goodbye true}}
+    {{if goodbye false}}
+    {{if goodbye true}}
+    {{if goodbye false}}
+    {{if goodbye true}}"
+>
+  Hello
+</div>
+<a href="/{{url}}/{{url}}"></a>
+`;
+
+exports[`element-modifier-statement.hbs 1`] = `
+<div {{hello param hash=key}} {{goodbye param}}>
+  Hello
+</div>
+
+<div {{hello param param param param param param param param param param param param}}>
+  Hello
+</div>
+
+<div {{hello hashPair=value hashPair=value hashPair=value hashPair=value hashPair=value}}>
+  Hello
+</div>
+
+<div {{hello param param param param hashPair=value hashPair=value hashPair=value hashPair=value hashPair=value}}>
+  Hello
+</div>
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+<div {{hello param hash=key}} {{goodbye param}}>
+  Hello
+</div>
+<div
+  {{hello
+    param
+    param
+    param
+    param
+    param
+    param
+    param
+    param
+    param
+    param
+    param
+    param
+  }}
+>
+  Hello
+</div>
+<div
+  {{hello
+    hashPair=value
+    hashPair=value
+    hashPair=value
+    hashPair=value
+    hashPair=value
+  }}
+>
+  Hello
+</div>
+<div
+  {{hello
+    param
+    param
+    param
+    param
+    hashPair=value
+    hashPair=value
+    hashPair=value
+    hashPair=value
+    hashPair=value
+  }}
+>
+  Hello
+</div>
+`;
+
+exports[`element-node.hbs 1`] = `
+<div class="attribute" {{modifier}} {{! comment}}>
+  Hello
+</div>
+
+<div>
+  Hello
+</div>
+
+<div>
+  hi
+</div>
+
+<div>
+  A long enough string to trigger a line break that would prevent wrapping.
+</div>
+
+<div>
+  A long enough string to trigger a line break that would prevent wrapping more.
+</div>
+
+<div>
+  A long enough string to trigger a line break that would prevent wrapping more and more.
+</div>
+
+<div>
+  {{#block}}
+    {{hello}}
+  {{/block}}
+</div>
+
+<div>
+  {{hello}}
+</div>
+
+<div></div>
+<img />
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+<div class="attribute" {{modifier}} {{! comment}}>
+  Hello
+</div>
+<div>
+  Hello
+</div>
+<div>
+  hi
+</div>
+<div>
+  A long enough string to trigger a line break that would prevent wrapping.
+</div>
+<div>
+  A long enough string to trigger a line break that would prevent wrapping more.
+</div>
+<div>
+  A long enough string to trigger a line break that would prevent wrapping more and more.
+</div>
+<div>
+  {{#block}}{{hello}}{{/block}}
```

Comment:
I don't understand the heuristics you chose. Sometimes it is in one line, sometimes it breaks, for things that look similar.

Reply:
I would prefer that HTML tags would break as frequently as block elements, but I remember running into some situations where that would cause some very ugly formatting. 

Perhaps block elements should also always break? What would be best?

Reply:
Here's a small example for JSX: [Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAeAJgSwG4D4A6UABEcAIYC+hqA9FngNzX27kW0tNQY5FmuUOOXFx7Y+RAEYD2dYaJZtgk2Z0IgANCAgAHGJmgBnZKDIAnMxADuABXMJjKMtgiZ0mkGUMxkAMzIANoZwWpJmZGAA1nAwAMo6EZhQAObIMGYAriEg6BBgfoHBWknBZjA24ckAtmQFQdkAVoYAHgBC4VExsWRVcAAySXB1RdoZMDpjAEzD2QlmpcggkmSSAJ4B0B46ZkkwAOpuMAAWyAAcAAxa2xDBe+E6i9twpdhDWmZwAI4ZmB8VZNVakh-PUtMEqpg0plsoYkskAnAAIoZCDwGZaGArA7oY7ISYY8KYAJwgDCECqNUWUGgbxAGWCABUVo4QcEKBQgA)

**Input:**
```jsx
<div>
  {a}
</div>;
<div>{a}</div>;
<div a>{a}</div>;
<div a b>{a}</div>;
<div>{a}{b}</div>;
```

**Output:**
```jsx
<div>{a}</div>;
<div>{a}</div>;
<div a>{a}</div>;
<div a b>
  {a}
</div>;
<div>
  {a}
  {b}
</div>;
```

It always break if there's >2 attributes or >2 values and inlines otherwise.

This is a small example but you should look at the JSX source code and see all the heuristics it does to make it look good.